Core/clim-core/frames.lisp

```diff
@@ -755,46 +756,60 @@ documentation produced by presentations.")
   (setf (slot-value pane 'name) name)
   pane)
 
-(defun do-pane-creation-form (name form)
-  (cond
-    ;; Single form which is a function call
-    ((and (= (length form) 1)
-          (listp (first form)))
-     `(coerce-pane-name ,(first form) ',name))
-    ;; Standard pane denoted by a keyword (i.e `:application')
-    ((keywordp (first form))
-     (case (first form)
-       (:application `(make-clim-application-pane
-                       :name ',name ,@(cdr form)))
-       (:interactor `(make-clim-interactor-pane
-                      :name ',name ,@(cdr form)))
-       (:pointer-documentation `(make-clim-pointer-documentation-pane
-                                 :name ',name ,@(cdr form)))
-       (:command-menu `(make-clim-command-menu-pane
-                                :name ',name ,@(cdr form)))
-       (otherwise `(make-pane ,(first form) :name ',name ,@(cdr form)))))
-    ;; Non-standard pane designator fed to the `make-pane'
-    (t `(make-pane ',(first form) :name ',name ,@(cdr form)))))
+(defun %generic-make-or-reinitialize-pane
+    (panes-for-layout abstract-class-name name &rest initargs)
+  (if-let ((pane (alexandria:assoc-value
+                  panes-for-layout name :test #'eq)))
+    (progn
+      (when-let ((parent (sheet-parent pane)))
+        (sheet-disown-child parent pane))
+      (apply #'reinitialize-instance pane initargs))
+    (apply #'make-pane abstract-class-name :name name initargs)))
+
+(defun make-ensure-pane-form (name form panes-for-layout-var)
```

Comment:
do we have a convention for functions which are used as helpers for macros? `do-pane-*` doesn't seem like a good convention either, but `make-*` seems to indicate that it is a constructor.

Reply:
`make-…-form` seems fine to me but I see your point. Would you prefer `generate-…-form`?

Reply:
generate makes more sense, yes

Reply:
Changed.